sonic_platform_base/sfp_base.py

```diff
@@ -304,3 +304,47 @@ def set_power_override(self, power_override, power_set):
             False if not
         """
         raise NotImplementedError
+
+    def read_eeprom_devid(self, port_num, devid, offset, num_bytes):
+        """
+        read eeprom specfic bytes for a given port_num and device id
+        begining from a random offset with size as num_bytes
+
+        Args:
+             port_num :
+                     Integer, index of physical port from where to read the bytes
+             devid :
+                     Integer, device id of the i2c bus required for reading for eeprom addr space
+             offset :
+                     Integer, the offset from which the read transaction will start
+             num_bytes:
+                     Integer, the number of bytes to be read
+
+        Returns:
+            a list , raw sequence of bytes which are read from the offset of size num_bytes
+        """
+        raise NotImplementedError
+
+    def write_eeprom_devid(self, port_num, devid, offset, num_bytes, write_buffer):
```

Comment:
There should be no need for port_num or devid here. If the platform uses sysfs/i2c to communicate with the EEPROM, the vendor will derive those based on which transceiver this function is called on and can in turn call the functions in sfputilbase.py.

Reply:
i agree port_num should come from SFP class instantiation, 
I think for devid it could only be 0x50 eeprom or 0x51 dom, 
But we can have fix this has 0x50 since we only have one i2c device to be read. 
So i will omit both these .

Reply:
fixed